src/tools/tidy/src/ui_tests.rs

```diff
@@ -135,6 +144,38 @@ pub fn check(path: &Path, bad: &mut bool) {
                     }
                 }
             }
+
+            if ext == "rs" {
+                lazy_static! {
+                    static ref ISSUE_NAME_REGEX: Regex =
+                        Regex::new(r"^issues?[-_]?(\d{3,})").unwrap();
+                }
+
+                if let Some(test_name) = ISSUE_NAME_REGEX.captures(testname) {
+                    // these paths are always relative to the passed `path` and always UTF8
+                    let stripped_path = file_path.strip_prefix(path).unwrap().to_str().unwrap();
+                    if !allowed_issue_filenames.remove(stripped_path) {
+                        tidy_error!(
+                            bad,
+                            "file `{stripped_path}` must begin with a descriptive name, consider `{{reason}}-issue-{issue_n}.rs`",
+                            issue_n = &test_name[1],
+                        );
+                    }
+                }
+            }
         }
     });
+
+    // if an excluded file is renamed, it must be removed from this list
+    if allowed_issue_filenames.len() > 0 {
+        for file_name in allowed_issue_filenames {
+            let mut p = PathBuf::from(path);
+            p.push(file_name);
+            tidy_error!(
+                bad,
```

Comment:
it would be very cool if tidy `--bless`ing removed it for you

Reply:
Hmm, that sounds plausible.